### PR TITLE
stop dependencies if some of them raised exception on start

### PR DESCRIPTION
**Pull request by bizywizy**

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -52,6 +52,50 @@ async def test_start_failed():
     assert service.running is False
 
 
+class Root(Simple):
+    def __init__(self, deps):
+        super(Root, self).__init__()
```
Comment (pohmelie) [inline test.py]: In modern python it is enough to use just `super().__init__()`

File: test.py [hunk 2]
```diff
@@ -52,6 +52,50 @@ async def test_start_failed():
...
+
+    assert valid.started is True
+    assert valid.stopped is True
+    assert valid.running is False
+
+
+class DelayedStartFailed(StartFailed):
+    async def start(self):
+        await asyncio.sleep(0)
+        await super(DelayedStartFailed, self).start()
```
Comment (pohmelie) [inline test.py]: Same here.

File: test.py [hunk 3]
```diff
@@ -52,6 +52,50 @@ async def test_start_failed():
...
+    valid = Simple()
+    broken = DelayedStartFailed()
+    root = Root([[broken, valid]])
+    assert root.running is False
+    with pytest.raises(RuntimeError):
+        await root.run()
+
+    assert valid.started is True
+    assert valid.stopped is True
+    assert valid.running is False
```
Comment (pohmelie) [inline test.py:110]: I'm not sure why `valid` is `started` and `stopped`, since it is next group. This test should fail, I think.

Comment (pohmelie) [reply test.py:110]: Even if start is delayed, next group wont start until previous starts ends.

Comment (pohmelie) [reply test.py:110]: Oh, I missed the double brackets. My bad. This fine, but add same test with single brackets where started and stopped will be `False`.

File: test.py (hunk 1 above)
Comment (bizywizy) [reply test.py]: Fixed

File: test.py (hunk 2 above)
Comment (bizywizy) [reply test.py]: Fixed

File: test.py (hunk 3 above)
Comment (bizywizy) [reply test.py:110]: Ok, I added it. Thanks.

**Comment by pohmelie** [comment]
@bizywizy thank you for your contribution! The new version `0.8.0` is on pypi since now.

**Pull request merged**